### DISCUSSION TITLE
Is it possible to extend the GraphQL API with custom resolvers?

### DISCUSSION THREAD

**Discussion by wspringer** [tinacms/tinacms#4531] [Help]
I keep struggling with image metadata. Next/image needs width and height, which makes perfect sense. TinaCMS should have access since it *knows* the images. One of the things I started wondering about is if there is a way to wrap the GraphQL resolver and redefine Image with something that has src, width and height. Is there a way for me to (1) get the default GraphQL API, and (2) augment it with my own resolvers? (I'm thinking graphql-tools wizardry.)

**Comment by jaschaio** [comment]
@wspringer did you find a way to do that?

I basically need this for adding [custom field components](https://tina.io/docs/extending-tina/custom-field-components/). It's not enough to just add a custom component if I can't define how the values get resolved within graphql.

**Reply by jaschaio** [reply]
Another use case: Let's say I define a block that fetches related posts (see https://github.com/tinacms/tinacms/discussions/4546) – without the ability to define a custom resolver I can only fetch the necessary data client side which is really bad.
